src/mmrelay/config.py

```
@@ -574,113 +759,221 @@ def apply_env_config_overrides(config: dict[str, Any] | None) -> dict[str, Any]:
-        else:
-            logger.error(f"Failed to create credentials.json at {credentials_path}")
+            # Verify the file was actually created
+            if os.path.exists(candidate):
+                logger.debug("Verified credentials.json exists at %s", candidate)
+            else:
+                logger.error("Failed to create credentials.json at %s", candidate)
+            return None
 
+        if last_error:
+            raise last_error
+        raise OSError("No candidate credentials paths available")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)